### ISSUE TITLE
[Suggestion] Follow a standard git branch practice

### ISSUE THREAD

**Issue by nisrulz** [eclipse/paho.mqtt.android#105]
So currently the `develop` branch is 3 commits behind the `master`.

I would like to propose a solution so that as devs who want to contribute can keep on doing so.

`master` - This branch should only contain stable code i.e releases included.

`develop` - This branch should be  the one where all the work should be going on. All pull requests should be made to `develop` branch. Once everything is done this can be merged into `master` branch.

`feature_*` - These are feature specific branches which devs can create from current `develop` branch and hence we able to send pull requests frequently with proper context.

Reference Link : http://nvie.com/posts/a-successful-git-branching-model/#the-main-branches

Let me know what are your thoughts on this.

**Comment by jpwsutton** [comment]
@nisrulz I believe that is already how we've been doing things since moving to GitHub. 

Those three commits were specific to the master branch that I had forgotten to make to the release-1.1.0 branch before it was merged in, so weren't appropriate to go into the develop branch as I would have had to undo them as soon as it was merged up. The changes were for the version numbers / instructions shown in the Readme, Issue template and the overall version definitions in the top level build.gradle file. Even if I made them all in one commit, the master would still need to be at least one commit ahead of develop.

There wouldn't be any changes to actual code though from now on in the master branch unless an urgent bugfix was needed, so hopefully the master branch can stay as is until the next release.
